### PR TITLE
DDP-6943: adding dnc checkbox and alert

### PR DESCRIPTION
**Pull request by SimoneMaiwald**
Signed-off-by: simone <simone@broadinstitute.org>

### CODE REVIEW

File: src/app/participant-page/participant-page.component.ts [hunk 1]
```diff
@@ -264,6 +270,28 @@ export class ParticipantPageComponent implements OnInit, OnDestroy {
...
+        },
+        err => {
+          this.openResultDialog( "Error - Failed to update participant" );
+        }
+      );
+      delete this.payload[ "data" ][ "doNotContact" ];
+    }
+    else {
+      this.updatedDNC = false;
+      debugger;
```
Comment (SimoneMaiwald) [inline src/app/participant-page/participant-page.component.ts]: updatedDNC is false when looking at the debugger… but checkbox is selected at the end

File: src/app/participant-page/participant-page.component.html [hunk 2]
```diff
@@ -136,7 +136,13 @@ <h3 class="Color--warn Line--Break">Lost to Follow-Up</h3>
           </tr>
           <tr>
             <td>Do Not Contact</td>
-            <td>{{getUtil().getYesNo(participant.data.profile['doNotContact'])}}</td>
+            <td *ngIf="hasRole().allowedToEditParticipant()">
+              <md-checkbox [value]="updatedDNC" color="primary" disableRipple
```
Comment (SimoneMaiwald) [inline src/app/participant-page/participant-page.component.html]: I tried [ngModel] [(ngModel)] [checked]
at the end it is always selected even if updatedDNC is false per debugger

File: src/app/participant-page/participant-page.component.html [hunk 3]
```diff
@@ -136,7 +136,13 @@ <h3 class="Color--warn Line--Break">Lost to Follow-Up</h3>
...
+                           (click)="updateDNC($event)"></md-checkbox>
```
Comment (SimoneMaiwald) [inline src/app/participant-page/participant-page.component.html]: I tried here (change) as well, but that didn't make a difference

File: src/app/participant-page/participant-page.component.html (hunk 2 above)
Comment (alextanlan) [reply src/app/participant-page/participant-page.component.html]: I believe it should be 
```
<md-checkbox [(ngModel)]="updatedDNC" color="primary" disableRipple
             (change)="updateDNC($event)">
</md-checkbox>
```

File: src/app/utils/utils.ts [hunk 4]
```diff
@@ -302,20 +300,7 @@ export class Utils {
             }
           }
         }
-        if (output.length > 1) {
-          let resultOutputSplitted = Utils.fillEmptyValuesFromCorrespondingOutputArray( output );
```
Comment (gmakharat) [inline src/app/utils/utils.ts:306]: I think those lines are responsible for multiple rows in CSV if participant has several records for chosen columns.

**Pull request closed without merging**